src/app/store/our.reducer.ts

```diff
@@ -0,0 +1,13 @@
+import { createReducer,on } from "@ngrx/store";
+import { event_updated } from "./our.actions";
+
+export const initialState:string[] = [];
+
+
+export const counterReducer = createReducer(
+  initialState,
+  on(event_updated,(state,payload)=>{let newstate= state.concat(payload.model.eventstring )
+  return newstate
```

Comment:
You have to replace the previous canvas data, rather than concatenating aa that data already contains all the information on the canvas.